sklearn/cluster/birch.py

```diff
@@ -336,23 +337,28 @@ class Birch(BaseEstimator, TransformerMixin, ClusterMixin):
     threshold : float, default 0.5
         The radius of the subcluster obtained by merging a new sample and the
         closest subcluster should be lesser than the threshold. Otherwise a new
-        subcluster is started.
+        subcluster is started. Setting this value to be very low, promotes
+        splitting and vice-versa.
 
     branching_factor : int, default 50
         Maximum number of CF subclusters in each node. If a new samples enters
         such that the number of subclusters exceed the branching_factor then
-        the node has to be split. The corresponding parent also has to be
-        split and if the number of subclusters in the parent is greater than
-        the branching factor, then it has to be split recursively.
+        that node is split into two nodes with the subclusters redistributed
+        in each. The parent subcluster of that node is removed and two new
+        subclusters are added as parents of the 2 split nodes.
 
     n_clusters : int, instance of sklearn.cluster model, default 3
         Number of clusters after the final clustering step, which treats the
-        subclusters from the leaves as new samples. If None, this final
-        clustering step is not performed and the subclusters are returned
-        as they are. If a model is provided, the model is fit treating
-        the subclusters as new samples and the initial data is mapped to the
-        label of the closest subcluster. If an int is provided, the model
-        fit is AgglomerativeClustering with n_clusters set to the int.
+        subclusters from the leaves as new samples.
+
+        "None" : the final clustering step is not performed and the subclusters are
```

Comment:
Also, I'd rather this be formatted using RST's list syntax (either bullet or definition) rather than individual paragraphs.